crates/runtime-table-partition/src/creator/filename.rs

```diff
@@ -0,0 +1,321 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//! Create filenames from [`ScalarValue`]s and infer [`ScalarValue`]s from
+//! filenames.
+//!
+//! `serde_qs` crate is used to serialize/deserialize [`ScalarValue`]s into
+//! [`String`]s that are URL encoded. This is helpful because we get the
+//! encoding and decoding functionality and special characters are escaped
+//! making them filesystem compatible.
+//!
+//! A subset of the [`ScalarValue`] variants are copied into a new type
+//! [`SupportedScalarValue`] so that we can derive `Serialize` and `Deserialize`
+//! on variants that have types that can be serialized/deserialized.
+
+use datafusion::scalar::ScalarValue;
+use serde::{Deserialize, Serialize};
+use snafu::prelude::*;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub))]
+pub enum Error {
+    #[snafu(display("Failed to serialize scalar value: {source}"))]
+    Serialize { source: serde_qs::Error },
+
+    #[snafu(display("Failed to deserialize scalar value: {source}"))]
+    Deserialize { source: serde_qs::Error },
+
+    #[snafu(display("Unsupported scalar value type"))]
+    UnsupportedType,
+}
+
+/// Converts a [`ScalarValue`] to its [`String`] representation.
+///
+/// # Errors
+/// Returns an error if the [`ScalarValue`] is not supported.
+pub fn encode_scalar_value(value: &ScalarValue) -> Result<String, Error> {
+    let supported_value = SupportedScalarValue::try_from(value.clone())?;
+    let encoded = serde_qs::to_string(&supported_value).context(SerializeSnafu)?;
+    Ok(encoded)
+}
+
+/// Converts a [`String`] back to a [`ScalarValue`].
+///
+/// # Errors
+/// Returns an error if a [`ScalarValue`] cannot be created.
+pub fn decode_scalar_value(value: &str) -> Result<ScalarValue, Error> {
+    let decoded: SupportedScalarValue = serde_qs::from_str(value).context(DeserializeSnafu)?;
+    let value = ScalarValue::try_from(decoded)?;
+    Ok(value)
+}
+
+#[derive(Serialize, Deserialize)]
+enum SupportedScalarValue {
+    Boolean(Option<bool>),
+    Int8(Option<i8>),
+    Int16(Option<i16>),
+    Int32(Option<i32>),
+    Int64(Option<i64>),
+    UInt8(Option<u8>),
+    UInt16(Option<u16>),
+    UInt32(Option<u32>),
+    UInt64(Option<u64>),
+    Utf8(Option<String>),
+    Utf8View(Option<String>),
+    LargeUtf8(Option<String>),
+    // TimestampSecond(Option<i64>, Option<String>), // TODO
```

Comment:
nit: remove or tracking ticket.

Reply:
Yes - we shouldn't leave `TODO` comments in the code (we got dinged on that for an audit). Either remove it or link to the issue where it will be soon implemented.